unit_tests/test_tlslite_extensions.py

```diff
@@ -1392,6 +1392,24 @@ def test_parse_with_extra_data_at_end(self):
         with self.assertRaises(SyntaxError):
             ext.parse(parser)
 
+    def test___repr__(self):
+        ext = SignatureAlgorithmsExtension().create([(HashAlgorithm.sha1,
+                                                      SignatureAlgorithm.rsa),
+                                                     (HashAlgorithm.sha256,
+                                                      SignatureAlgorithm.rsa),
+                                                     (HashAlgorithm.sha384,
+                                                      SignatureAlgorithm.dsa)])
+
+        self.assertEqual(repr(ext),
+                "SignatureAlgorithmsExtension("
+                "sigalgs=[rsa_pkcs1_sha1, rsa_pkcs1_sha256, (sha384, dsa)])")
```

Comment:
Those `rsa_pkcs1_sha1` names already come from TLS 1.3, and yes, it's only to provide something that is human readable, not something that is machine readable, so free form approach is fine.
Finally, if it cannot translate a number to a name (because it is undefined), it will simply return a number as a string